nsdb-sql/src/main/scala/io/radicalbit/nsdb/sql/parser/SQLStatementParser.scala

```diff
@@ -104,26 +104,36 @@ final class SQLStatementParser extends RegexParsers with PackratParsers with Reg
   private val CloseRoundBracket = ")"
   private val temporalInterval  = "INTERVAL" ignoreCase
 
-  private val digits           = """(^(?!now)[a-zA-Z_][a-zA-Z0-9_]*)""".r
-  private val digitsWithDashes = """(^(?!now)[a-zA-Z_][a-zA-Z0-9_\-]*[a-zA-Z0-9]*)""".r
-  private val numbers          = """([0-9]+)""".r
-  private val intValue         = numbers ^^ { _.toInt }
-  private val longValue        = numbers ^^ { _.toLong }
-  private val doubleValue      = """([0-9]+)\.([0-9]+)""".r ^^ { _.toDouble }
-
-  private val field = digits ^^ { e =>
+  private val letter          = """[a-zA-Z_]"""
+  private val digit           = """[0-9]"""
+  private val letterOrDigit   = """[a-zA-Z0-9_]"""
+  private val specialChar     = """[a-zA-Z0-9_\-\.:~]"""
+  private val specialWildCard = """[a-zA-Z0-9_\-$\.:~]"""
+
+  private val standardString = s"""($letter$letterOrDigit*)""".r
+  private val specialString  = s"""($letter$specialChar*$letterOrDigit*)""".r
+  private val wildCardString = s"""($specialWildCard+)""".r
+  private val digits         = s"""($digit+)""".r
+  private val intValue       = digits ^^ { _.toInt }
+  private val longValue      = digits ^^ { _.toLong }
+  private val doubleValue    = s"""($digit+)[.]($digit+)""".r ^^ { _.toDouble }
+
+  private val field = standardString ^^ { e =>
     Field(e, None)
   }
-  private val aggField = ((sum | min | max | count | first | last | avg) <~ OpenRoundBracket) ~ (digits | All) <~ CloseRoundBracket ^^ {
-    e =>
-      Field(e._2, Some(e._1))
+  private val aggField = ((sum | min | max | count | first | last | avg) <~ OpenRoundBracket) ~ (standardString | All) <~ CloseRoundBracket ^^ {
+    case aggregation ~ name => Field(name, Some(aggregation))
   }
-  private val dimension = digits
-  private val stringValue = (digitsWithDashes | (("'" ?) ~> (digitsWithDashes +) <~ ("'" ?))) ^^ {
+  private val dimension = standardString
```

Comment:
why this assignment?

Reply:
It's just for syntactic sugar. To make any grammar rule containing dimensions (e.g. comparison expressions) more readable. Not sure if this makes sense

Reply:
okay, got it. Up for me!